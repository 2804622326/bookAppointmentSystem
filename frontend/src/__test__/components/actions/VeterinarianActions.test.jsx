import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import VeterinarianActions from '../../../components/actions/VeterinarianActions.jsx';

const appointment = { id: 1 };

test('renders VeterinarianActions and handles approve/decline', () => {
  const onApprove = jest.fn(() => Promise.resolve());
  const onDecline = jest.fn(() => Promise.resolve());
  const { asFragment } = render(
    <VeterinarianActions
      onApprove={onApprove}
      onDecline={onDecline}
      isDisabled={false}
      appointment={appointment}
    />
  );

  const approveButton = screen.getByRole('button', { name: /approve appointment/i });
  const declineButton = screen.getByRole('button', { name: /decline appointment/i });

  expect(approveButton).toBeInTheDocument();
  expect(declineButton).toBeInTheDocument();

  fireEvent.click(approveButton);
  expect(onApprove).toHaveBeenCalledWith(1);

  fireEvent.click(declineButton);
  expect(onDecline).toHaveBeenCalledWith(1);

  expect(asFragment()).toMatchSnapshot();
});
