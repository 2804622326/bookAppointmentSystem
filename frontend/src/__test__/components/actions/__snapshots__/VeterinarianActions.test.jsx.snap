// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VeterinarianActions and handles approve/decline 1`] = `
<DocumentFragment>
  <section
    class="d-flex justify-content-end gap-2 mt-2 mb-2"
  >
    <button
      class="btn btn-success btn-sm"
      type="button"
    >
      Approve Appointment
    </button>
    <button
      class="btn btn-secondary btn-sm"
      type="button"
    >
      <div
        class="text-center"
      >
        <span
          aria-hidden="true"
          class="spinner-border spinner-border-sm"
          role="status"
        />
        <span
          class="sr-only"
        >
          Declining appointment...
        </span>
      </div>
    </button>
  </section>
</DocumentFragment>
`;
